tools/chapel-py/chapel.cpp

```diff
@@ -0,0 +1,751 @@
+PLAIN_GETTER(StringLikeLiteral, value, "s", return node->value());
+
+PLAIN_GETTER(Function, kind, "s", return chpl::uast::Function::kindToString(node->kind()));
+PLAIN_GETTER(Function, is_method, "b", return node->isMethod());
+PLAIN_GETTER(Function, is_primary_method, "b", return node->isPrimaryMethod());
+PLAIN_GETTER(Function, this_formal, "O", return wrapAstNode(contextObject, node->thisFormal()));
+PLAIN_GETTER(Function, formals, "O", return wrapIterPair(contextObject, node->formals()));
+PLAIN_GETTER(FnCall, used_square_brackets, "b", return node->callUsedSquareBrackets());
+PLAIN_GETTER(FnCall, called_expression, "O", return wrapAstNode(contextObject, node->calledExpression()));
+PLAIN_GETTER(OpCall, op, "s", return node->op());
+PLAIN_GETTER(Dot, field, "s", return node->field());
+PLAIN_GETTER(VisibilityClause, symbol, "O", return wrapAstNode(contextObject, node->symbol()));
+PLAIN_GETTER(Identifier, name, "s", return node->name());
+PLAIN_GETTER(SimpleBlockLike, block_style, "s", return blockStyleToString(node->blockStyle()));
+PLAIN_GETTER(Loop, block_style, "s", return blockStyleToString(node->blockStyle()));
+PLAIN_GETTER(EnumElement, init_expression, "O", return wrapAstNode(contextObject, node->initExpression()));
+
+
+static PyObject* AttributeObject_actuals(PyObject *self, PyObject *Py_UNUSED(ignored)) {
+  auto attributeNode = ((AttributeObject*) self)->parent.astNode->toAttribute();
+
+  auto argList = Py_BuildValue("(O)", (PyObject*) self);
+  auto astCallIterObjectPy = PyObject_CallObject((PyObject *) &AstCallIterType, argList);
+  Py_XDECREF(argList);
+
+  auto astCalliterObject = (AstCallIterObject*) astCallIterObjectPy;
+  astCalliterObject->current = 0;
+  astCalliterObject->num = attributeNode->numActuals();
+  astCalliterObject->container = attributeNode;
+  astCalliterObject->childGetter = [](const void* node, int child) {
+    return ((chpl::uast::Attribute*) node)->actual(child);
+  };
+  astCalliterObject->nameGetter = [](const void* node, int child) {
+    return ((chpl::uast::Attribute*) node)->actualName(child);
+  };
+
+  return astCallIterObjectPy;
+}
+
+static PyObject* FnCallObject_actuals(PyObject *self, PyObject *Py_UNUSED(ignored)) {
+  auto fnCallNode = ((FnCallObject*) self)->parent.astNode->toFnCall();
+
+  auto argList = Py_BuildValue("(O)", (PyObject*) self);
+  auto astCallIterObjectPy = PyObject_CallObject((PyObject *) &AstCallIterType, argList);
+  Py_XDECREF(argList);
+
+  auto astCalliterObject = (AstCallIterObject*) astCallIterObjectPy;
+  astCalliterObject->current = 0;
+  astCalliterObject->num = fnCallNode->numActuals();
+  astCalliterObject->container = fnCallNode;
+  astCalliterObject->childGetter = [](const void* node, int child) {
+    return ((chpl::uast::FnCall*) node)->actual(child);
+  };
+  astCalliterObject->nameGetter = [](const void* node, int child) {
+    return ((chpl::uast::FnCall*) node)->actualName(child);
+  };
+
+  return astCallIterObjectPy;
+}
```

Comment:
It would be nice to generalize this with a macro pattern. Definitely more of a feature request, just a thought

Reply:
I suspect there will only be two things with actuals, so generalizing seemed premature. Might help with code re-use, though.